### PR TITLE
Fix unique key in ui-notification-toast

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Notifications

### Description
Fix unique key error in ui notification

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

**Review by ErikMichelson — APPROVED** [review]

**Review by ErikMichelson — APPROVED** [review]

File: src/components/notifications/ui-notification-toast.tsx [hunk 1]
```diff
@@ -74,13 +74,13 @@ export const UiNotificationToast: React.FC<UiNotificationProps> = ({
...
     () =>
-      buttons?.map((button) => {
+      buttons?.map((button, buttonIndex) => {
         const buttonClick = () => {
           button.onClick()
           dismissThisNotification()
         }
         return (
-          <Button key={button.label} size={'sm'} onClick={buttonClick} variant={'link'}>
+          <Button key={buttonIndex} size={'sm'} onClick={buttonClick} variant={'link'}>
```
Comment (DerMolly) [inline src/components/notifications/ui-notification-toast.tsx:83]: Maybe a prefix would be usefull here? Something like `notification-${buttonIndex}`

File: src/components/notifications/ui-notification-toast.tsx [hunk 2]
```diff
@@ -89,9 +89,9 @@ export const UiNotificationToast: React.FC<UiNotificationProps> = ({
   )
 
   const contentDom = useMemo(() => {
-    return content.split('\n').map((value) => {
+    return content.split('\n').map((value, lineNumber) => {
       return (
-        <Fragment>
+        <Fragment key={lineNumber}>
```
Comment (DerMolly) [inline src/components/notifications/ui-notification-toast.tsx:94]: Same here. Additionally doesn't this break if there are more than one notification on the screen at the same time?

File: src/components/notifications/ui-notification-toast.tsx (hunk 1 above)
Comment (mrdrogdrog) [reply src/components/notifications/ui-notification-toast.tsx:83]: Why? Keys only need to be unique within their parent.

File: src/components/notifications/ui-notification-toast.tsx (hunk 2 above)
Comment (mrdrogdrog) [reply src/components/notifications/ui-notification-toast.tsx:94]: s.o.

**Pull request merged**